### DISCUSSION TITLE
Custom Block in Nuxt, how to use them?

### DISCUSSION THREAD

**Discussion by casualmatt** [nuxt/nuxt#20687] [Questions]
Hello to all, 

When I started using custom blocks inside the i18n module, I understood their value for low technical users who need to maintain a nuxt project easily and with minimal knowledge of JS/TS.

But, finding documentation on SFC Custom Block was pretty hard; with some hint from @danielroe in live found out that vue auto export custom block in the order funded on the page: block0, block1, ...

Following what the i18n module has done, I tried to rename the export like that, but Vite crashed without a proper error message.
https://stackblitz.com/edit/nuxt-starter-yhmkdz?file=app.vue

I'm open to suggestions to improving my example so that it could work; ty for your time. 

Matt

**Comment by codywakeford** [comment]
This does the trick.

nuxt.config.ts/ vue.config.ts
```ts 

build: {
        extend(config, { isDev, isClient }) {
            const vueLoader = config.module.rules.find((rule) => rule.loader === "vue-loader")

            if (vueLoader) {
                // Extend Vue loader to recognize your custom blocks
                vueLoader.options.compilerOptions = {
                    customBlocks: ["custom-block"], // List your custom block names here
                }
            }
        },
    } ```

**Reply by casualmatt** [reply]
Hi @codywakeford,

Thanks for the reply, but that's for a custom block in Webpack. But I'm unsure if that code even works in Webpack, as the documentation I found describes something completely different.

**Reply by husayt** [reply]
add a vite plugin like this:
```
import type { Plugin } from 'vite'

export const vueI18nPlugin: Plugin = {
  name: 'vue-i18n',
  async transform(code, id) {
    if (!/vue&type=i18n/.test(id)) {
      return
    }
    if (/\.ya?ml$/.test(id)) {
      const { load } = await import('js-yaml')
      code = JSON.stringify(load(code.trim()))
    }
    return {
      code: `export default Comp => {
      Comp.i18n = ${code}
    }`,
      map: { mappings: '' },
    }
  },
}
```
